public/apps/configuration/top-nav-menu.tsx

```diff
@@ -59,6 +59,7 @@ export const SecurityPluginTopNavMenu = React.memo(
               : undefined,
           onSelectedDataSources: wrapSetDataSourceWithUpdateUrl,
           fullWidth: true,
+          dataSourceFilter: (ds) => ds.attributes.auth.credentials?.service !== 'aoss',
```

Comment:
Adding a test is overkill for this plugin. OSD core has a test: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b422791347c22029f0cf155ff24492d0ea9e3c8d/src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.test.tsx#L86, and we should not bog ourselves down with internal implementations.

Reply:
What are the possible values for service?

Reply:
The values are 'aoss', or 'es', but good call out here. I pulled it from core as a constant since it is available. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/a2ee3629040c71bcf1d5d02a0c110703b4e5ca44/src/plugins/data_source/common/data_sources/types.ts#L50

Reply:
Should we flip this around and only show datasource that is OpenSearch? Will there be other types of datasources in the future?

@RyanL1997 perhaps you know if other datasource types will be added in a future release?

Reply:
We actually cannot import the constant as core does not export it. I will create an issue to do that, but reverted the change for now.

Reply:
yes ideally we will add more types of datasources in the future. (cc @derek-ho @cwperks )